File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        MIN = float('inf')
        n = len(nums)
        if n == 1:return 0
        if k >= 2:
            nums.sort()
            for i in range(n-1):
                diff = nums[i+1] - nums[i]
                if diff < MIN:
                    MIN = diff
            return MIN                       

File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        int MIN = INT_MAX;
        int n = nums.size();
        if(n == 1 || k == 1) return 0;
        sort(nums.begin(), nums.end());
        int i = 0 ,j = k;
        while(j <= n){
            long mx = nums[j-1];
            long mn = nums[i];
            long diff = mx - mn;
            if(diff < MIN) MIN = diff;
            i++;
            j++;
        }
        return MIN;
    }
};